src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -1261,150 +1288,81 @@ def serialize_with_spans(
+                parent_error_map[error["trace.transaction"]].append(self.serialize_error(error))
+            else:
+                orphan_errors.append(error)
 
-            # Determine orphan transactions. `trace.parent_transaction` on a
-            # transaction is set when the indexed spans dataset has a row for
-            # the parent span id for this transaction. Since we already considered
-            # the root spans cases, the remaining spans with no parent transaction
-            # id are orphan transactions.
-            orphan_roots = [
-                orphan
-                for orphan in remaining_transactions
-                if orphan["trace.parent_transaction"] is None
-            ]
-
-            # Build the trees for all the orphan transactions.
-            orphans = self.visit_transactions(
-                orphan_roots,
-                remaining_transactions,
-                errors,
-                visited_transactions,
-                visited_errors,
-            )
+        with sentry_sdk.start_span(op="serialize", description="associate children"):
+            visited_transactions: set[str] = {
+                transaction.event["id"] for transaction in root_traces
+            }
+            for transaction in serialized_transactions:
+                event_id = transaction.event["id"]
+                if event_id in parent_event_map:
+                    children_events = parent_event_map.pop(event_id)
+                    transaction.children = sorted(children_events, key=child_sort_key)
+                if event_id in parent_error_map:
+                    transaction.errors = sorted(
+                        parent_error_map.pop(event_id), key=lambda k: k["timestamp"]
+                    )
 
-        with sentry_sdk.start_span(op="nodestore", description="remaining orphans"):
-            # Remaining are transactions with parent transactions but those
-            # parents don't map to any of the existing transactions.
-            remaining_transactions = self.calculate_remaining_transactions(
-                transactions, visited_transactions
-            )
-            orphans.extend(
-                self.visit_transactions(
-                    remaining_transactions,
-                    remaining_transactions,
-                    errors,
-                    visited_transactions,
-                    visited_errors,
-                )
-            )
+        with sentry_sdk.start_span(op="serialize", description="more orphans"):
+            for transaction in sorted(serialized_transactions, key=child_sort_key):
+                if transaction.event["id"] not in visited_transactions:
+                    if transaction not in orphans:
+                        orphans.append(transaction)
+                    visited_transactions.add(transaction.event["id"])
+                    for child in transaction.children:
+                        visited_transactions.add(child.event["id"])
```

Comment:
You're basically just using visited_transactions for this block to only visit each serialized transaction once so I'd relocate it here